.github/workflows/linux-workflow.yaml

```diff
@@ -105,9 +103,27 @@ jobs:
         continue-on-error: true
         with:
           project-token: ${{ secrets.CODACY_PROJECT_KEY }}
-          # lcov report
           coverage-reports: ./build/goalc-test_coverage.info
 
+      - name: Upload artifact
+        # NOTE - the GCC executables are huge as they've been compiled with coverage support
+        if: matrix.compiler == 'clang'
+        uses: actions/upload-artifact@v3
+        with:
+          name: opengoal-${{ matrix.os }}-${{ matrix.compiler }}
+          path: |
+            ./build/**/*.so
+            ./build/**/*.a
+            ./build/decompiler/decompiler
+            ./build/decompiler/extractor
+            ./build/game/gk
+            ./build/sound/sndplay
+            ./build/goalc/goalc
+            ./build/tools/dgo_packer
+            ./build/tools/dgo_unpacker
+            ./build/tools/memory_dump_tolls
+            ./build/tools/level_tools/level_dump
+
```

Comment:
```suggestion
            build/**/*.so
            build/**/*.a
            build/decompiler/decompiler
            build/decompiler/extractor
            build/game/gk
            build/sound/sndplay
            build/goalc/goalc
            build/tools/dgo_packer
            build/tools/dgo_unpacker
            build/tools/memory_dump_tolls
            build/tools/level_tools/level_dump
        if-no-files-found: error

```

Reply:
The artifacts are just a nice to have, i don't want to fail the build over them not being found.  I also prefer the leading `./` style, im aware it's not required.